src/keymgmt.c

```diff
@@ -482,18 +482,30 @@ static int p11prov_common_gen(struct key_generator *ctx,
     }
 
     ret = p11prov_merge_pub_attrs_into_priv(pub_key, priv_key);
```

Comment:
perhaps we can remove this merging, given that now we can always carry the public key linked

Reply:
No need to do it in this PR but you may add a comment to the effect that this call is redundant now that we set a public key

Reply:
Added `TODO` to `p11prov_merge_pub_attrs_into_priv` definition